src/uattributes/uattributesvalidator.rs

```diff
@@ -556,29 +558,30 @@ mod tests {
+    #[test_case(UMessageType::UMESSAGE_TYPE_PUBLISH, None, None, Some(500), false; "for Publish message without ID with TTL")]
+    #[test_case(UMessageType::UMESSAGE_TYPE_PUBLISH, Some(1000), Some(UUIDBuilder::build()), None, false; "for Publish message with ID without TTL")]
+    #[test_case(UMessageType::UMESSAGE_TYPE_PUBLISH, Some(1000), Some(UUIDBuilder::build()), Some(0), false; "for Publish message with ID and TTL 0")]
+    #[test_case(UMessageType::UMESSAGE_TYPE_PUBLISH, Some(1000), Some(UUIDBuilder::build()), Some(500), true; "for Publish message with ID and expired TTL")]
+    #[test_case(UMessageType::UMESSAGE_TYPE_PUBLISH, Some(1000), Some(UUIDBuilder::build()), Some(2000), false; "for Publish message with ID and non-expired TTL")]
+    #[test_case(UMessageType::UMESSAGE_TYPE_REQUEST, None, None, None, false; "for Request message without ID nor TTL")]
+    #[test_case(UMessageType::UMESSAGE_TYPE_REQUEST, None, None, Some(0), false; "for Request message without ID with TTL 0")]
+    #[test_case(UMessageType::UMESSAGE_TYPE_REQUEST, None, None, Some(500), false; "for Request message without ID with TTL")]
+    #[test_case(UMessageType::UMESSAGE_TYPE_REQUEST, Some(1000), Some(UUIDBuilder::build()), None, false; "for Request message with ID without TTL")]
+    #[test_case(UMessageType::UMESSAGE_TYPE_REQUEST, Some(1000), Some(UUIDBuilder::build()), Some(0), false; "for Request message with ID and TTL 0")]
+    #[test_case(UMessageType::UMESSAGE_TYPE_REQUEST, Some(1000), Some(UUIDBuilder::build()), Some(500), true; "for Request message with ID and expired TTL")]
+    #[test_case(UMessageType::UMESSAGE_TYPE_REQUEST, Some(1000), Some(UUIDBuilder::build()), Some(2000), false; "for Request message with ID and non-expired TTL")]
+    #[test_case(UMessageType::UMESSAGE_TYPE_RESPONSE, None, None, None, false; "for Response message without ID nor TTL")]
+    #[test_case(UMessageType::UMESSAGE_TYPE_RESPONSE, None, None, Some(0), false; "for Response message without ID with TTL 0")]
+    #[test_case(UMessageType::UMESSAGE_TYPE_RESPONSE, None, None, Some(500), false; "for Response message without ID with TTL")]
+    #[test_case(UMessageType::UMESSAGE_TYPE_RESPONSE, Some(1000), Some(UUIDBuilder::build()), None, false; "for Response message with ID without TTL")]
+    #[test_case(UMessageType::UMESSAGE_TYPE_RESPONSE, Some(1000), Some(UUIDBuilder::build()), Some(0), false; "for Response message with ID and TTL 0")]
+    #[test_case(UMessageType::UMESSAGE_TYPE_RESPONSE, Some(1000), Some(UUIDBuilder::build()), Some(500), true; "for Response message with ID and expired TTL")]
+    #[test_case(UMessageType::UMESSAGE_TYPE_RESPONSE, Some(1000), Some(UUIDBuilder::build()), Some(2000), false; "for Response message with ID and non-expired TTL")]
     fn test_is_expired(
         message_type: UMessageType,
+        created_millis_ago: Option<u64>,
```

Comment:
We don't need Option here and only sleep while the created_millis_ago is not 0.

Reply:
Good point!